example/create_dependencies.py

```diff
@@ -74,6 +87,19 @@ def usage():
     # pylint: disable=invalid-name
     all_files = sys.argv[1:]
 
+    # Create a look-up directories that maps the filenames without
+    # path and extensions to the full filename:
+    lookup_files = {}
+    for file in all_files:
+        root = get_root(file)
+        if root in lookup_files:
+            print(
+                f"The file '{file}' has the same root '{root}' as "
+                f"the file '{lookup_files[root]}'."
+            )
+            sys.exit(-1)
+        lookup_files[get_root(file)] = file
```

Comment:
You can just use `root` here rather than doing `get_root(file)` I think?

Reply:
Indeed.